ext/symengine/ruby_real_mpfr.c

```diff
@@ -1,7 +1,22 @@
 #include "ruby_real_mpfr.h"
 
-VALUE crealmpfr_init(VALUE self, VALUE num_value) {
-    
-    return self;
+VALUE crealmpfr_init(VALUE self, VALUE num_value, VALUE prec_value) {
+    basic_struct *cresult;
+    VALUE result;
+    double d;
+    int prec = NUM2INT(prec_value);
+    cresult = basic_new_heap();
+
+    switch( TYPE(num_value) ) {
+        case T_FLOAT:
+            d = RFLOAT_VALUE(num_value);     
+            real_mpfr_set_d(cresult, d, prec);
+            break;
+        default:
+            rb_raise(rb_eTypeError, "Invalid Type: Float, BigDecimal or String required.");
+            break;
+    }
+    result = Data_Wrap_Struct(c_real_mpfr, NULL, cbasic_free_heap, cresult);
+    return result;
```

Comment:
Thanks @isuruf got it to be working, only to proceed into another confusion.

I'm getting the following output. Should I be worried?

```
2.1.8 :012 > X = SymEngine::RealMPFR.new(123.456, 200)
 => #<SymEngine::RealMPFR(123.45600000000000306954461848363280296325683593750000000000000)> 
2.1.8 :013 > Y = SymEngine::RealMPFR.new(456.123, 200)
 => #<SymEngine::RealMPFR(456.12299999999999045030563138425350189208984375000000000000000)> 
```

I am guessing this is to be expected with the arbitrary precision. But just wanted to confirm.